ScrapPackedLibrary/ScrapPackedFile.cs

```diff
@@ -243,53 +242,78 @@ private void ExtractFile(string p_packedPath, string p_destinationPath, FileStre
 
             var fsPacked = p_PackedFileStream;
-            if(fsPacked==null)
+            if (fsPacked == null)
                 fsPacked = new FileStream(fileName, FileMode.Open);
+
+            var fsExtractFile = new FileStream(p_destinationPath, FileMode.Create);
             try
             {
-                var fsExtractFile = new FileStream(p_destinationPath, FileMode.Create);
-                try
-                {
-                    byte[] readBytes = new byte[fileMetaData.FileSize];
+                byte[] readBytes = new byte[fileMetaData.FileSize];
 
-                    fsPacked.Seek(fileMetaData.OriginalOffset, SeekOrigin.Begin);
-                    fsPacked.Read(readBytes, 0, (int)fileMetaData.FileSize);
+                fsPacked.Seek(fileMetaData.OriginalOffset, SeekOrigin.Begin);
+                fsPacked.Read(readBytes, 0, (int)fileMetaData.FileSize);
                     
-                    fsExtractFile.Write(readBytes);
-                }
-                finally
-                {
-                    fsExtractFile.Close();
-                }
+                fsExtractFile.Write(readBytes);
             }
             finally
             {
-                if (p_PackedFileStream == null)
-                    fsPacked.Close();
+                fsExtractFile.Close();
             }
-
-
+            if (p_PackedFileStream == null)
+                fsPacked.Close();
         }
 
         public void SaveToFile(string p_newFileName)
         {
             metaData.RecalcFileOffsets();
 
             var newFileName = fileName;
+            var oldFileName = fileName;
+
             if (p_newFileName.Length > 0)
                 newFileName = p_newFileName;
             else
             {
                 if (File.Exists(fileName))
-                    File.Move(fileName, fileName + ".bak",true);
+                    File.Move(fileName, fileName + ".bak", true);
                 fileName = fileName + ".bak";
-            }   
+            }
 
             if (File.Exists(newFileName))
                 File.Delete(newFileName);
 
             // todo: make backup function better
-            Directory.CreateDirectory(Path.GetDirectoryName(newFileName));
+
+            string dirName = Path.GetDirectoryName(newFileName);
+            
+            if (dirName == null)
+            {
+                // restore backup
+                if (File.Exists(fileName))
+                    File.Move(fileName, oldFileName, true);
+                fileName = oldFileName;
+
+                throw new IOException($"Unable to save file {p_newFileName}: unexpected error.");
+            }
+            else if (dirName != "") // if dirName is not same dir as working dir. 
+            { 
+                try
+                {
+                    // todo: when input "-o <folder_name>/<file_name>" while file (NOT DIR!)
+                    // with name <folder_name> already existing couse excteption. 
```

Comment:
Can you explain this todo more?
there can be a folder and a file with the same name, right?

also multiple typos: "cause exception" or "raise exception"

Reply:
No, there is can't be folder and a file with a same name, including extension. 
This is no more bug because exceptions are already handling properly